validator/src/lib.rs

```diff
@@ -1,30 +1,33 @@
 extern crate valico;
 extern crate serde_json;
-
-use std::error::Error;
+#[macro_use] extern crate error_chain;
 
 use serde_json::{Value};
 use valico::json_schema;
 
-pub fn get_schema(api_version: &str) -> Result<Value, Box<Error>> {
+pub mod errors;
+
+use errors::*;
+
+pub fn get_schema(api_version: &str) -> Result<Value> {
     let schema = match api_version {
         "0.13" => include_str!("../schema/13.json"),
         "0.12" => include_str!("../schema/12.json"),
         "0.11" => include_str!("../schema/11.json"),
         "0.9" => include_str!("../schema/9.json"),
         "0.8" => include_str!("../schema/8.json"),
-        _ => return Err(format!("Invalid api version: {}", api_version).into())
-    }.parse()?;
+        _ => return Err(ErrorKind::WrongApiVersion(api_version.into()).into())
```

Comment:
The first is for the `str` into `String`, the second I'm not sure, need to check :wink: